browser/brave_stats_updater.cc

```diff
@@ -0,0 +1,174 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_stats_updater.h"
+
+#include "base/sys_info.h"
+#include "brave/browser/brave_stats_updater_params.h"
+#include "brave/common/pref_names.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/net/system_network_context_manager.h"
+#include "chrome/common/channel_info.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/version_info/version_info.h"
+#include "net/base/load_flags.h"
+#include "net/base/url_util.h"
+#include "net/traffic_annotation/network_traffic_annotation.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+
+const char kBaseUpdateURL[] = "https://laptop-updates.brave.com/1/usage/brave-core";
+
+const char kPlatformIdentifierLinux[] = "linux-bc";
+const char kPlatformIdentifierMac[] = "osx-bc";
+const char kPlatformIdentifierWinIA32[] = "winia32-bc";
+const char kPlatformIdentifierWinx64[] = "winx64-bc";
+
+// Ping the update server once an hour.
+const int kUpdateServerPingFrequency = 60 * 60;
+
+// Maximum size of the server ping response in bytes.
+const int kMaxUpdateServerPingResponseSizeBytes = 1024 * 1024;
+
+namespace {
+
+std::string GetChannelName() {
+  std::string channel = chrome::GetChannelName();
```

Comment:
@bbondy I suspect I'm not doing the right thing here. The update server supports channels of 'dev', 'release', 'nightly', and 'beta'. Are those available to me in brave-core?

Reply:
@simonhong to give the correct names, we'll have to map it.

Reply:
In brave-core, `canary` string is used in many other places.
So, if we want to use another string like `nightly` instead of `canary`,
I recommend to make mapping function like `GetChannelName(version_info::Channel channel)` here.
Does this way work?

Reply:
After discussion with @bbondy , we will replace `canary` with `nightly` in brave-core.